src/codegeneration/ES6ClassTransformer.js

```diff
@@ -0,0 +1,261 @@
+// Copyright 2015 Traceur Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the 'License');
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an 'AS IS' BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {
+  CONSTRUCTOR
+} from '../syntax/PredefinedName.js';
+import {
+  AnonBlock,
+  ClassExpression,
+  GetAccessor,
+  PropertyMethodAssignment,
+  SetAccessor,
+  LiteralPropertyName,
+} from '../syntax/trees/ParseTrees.js';
+import {
+  CALL_EXPRESSION,
+  EXPRESSION_STATEMENT,
+  GET_ACCESSOR,
+  PROPERTY_METHOD_ASSIGNMENT,
+  PROPERTY_VARIABLE_DECLARATION,
+  SET_ACCESSOR,
+  SUPER_EXPRESSION
+} from '../syntax/trees/ParseTreeType.js';
+import {TempVarTransformer} from './TempVarTransformer.js';
+import {ParseTreeTransformer} from './ParseTreeTransformer.js';
+import {
+  createBindingIdentifier,
+  createClassDeclaration,
+  createCommaExpression,
+  createEmptyParameterList,
+  createExpressionStatement,
+  createFunctionBody,
+  createIdentifierToken,
+  createImmediatelyInvokedFunctionExpression,
+  createPropertyNameAssignment,} from './ParseTreeFactory.js';
+import {parseStatement, parseExpression} from './PlaceholderParser.js';
+import {propName} from '../staticsemantics/PropName.js';
+import {prependStatements} from './PrependStatements.js';
+
+// TODO(vicb): comment
+
+export class ES6ClassTransformer extends TempVarTransformer {
+  /**
+   * @param {UniqueIdentifierGenerator} identifierGenerator
+   */
+  constructor(identifierGenerator) {
+    super(identifierGenerator);
+  }
+
+  transformClassElements_(tree) {
+    let superClass = this.transformAny(tree.superClass);
+
+    let elements = [];
+    let initInstanceVars = [], initStaticVars = [];
+    let constructor;
+    let constructorIndex = 0;
+
+    tree.elements.forEach((tree) => {
+      let initVars;
+      if (tree.isStatic) {
+        initVars = initStaticVars;
+      } else {
+        initVars = initInstanceVars;
+      }
+
+      switch (tree.type) {
+        case GET_ACCESSOR:
+          let getter = new GetAccessor(tree.location, tree.isSatic, tree.name,
+              tree.typeAnnotation, tree.annotations, tree.body);
+          elements.push(this.transformAny(getter));
+          break;
+
+        case SET_ACCESSOR:
+          let setter = new SetAccessor(tree.location, tree.isSatic, tree.name,
+              tree.parameterList, tree.annotations, tree.body);
+          elements.push(this.transformAny(setter));
+          break;
+
+        case PROPERTY_METHOD_ASSIGNMENT:
+          if (!tree.isStatic && propName(tree) === CONSTRUCTOR) {
```

Comment:
Maybe not actually, in V8 I check this in the parser but I think I forgot to add these checks to Traceur so Traceur is wrong to parse such a class.

I filed #1793 so you do not have to care about this at this stage.